docs: added Cloudflare R2 guide for Object Storage (#10497)
medusajs/medusa: www/apps/resources/app/architectural-modules/file/s3/page.mdx

      4. Click ALL for "Allowed Operations", click the "Review" button, then the "Save policy" button.

    </TabsContent>
    <TabsContent value="Cloudflare R2">

>     1. Create a [Cloudflare free account](https://supabase.com/dashboard/sign-in) if you don’t have.

Reviewer: ```suggestion
    1. Create a [Cloudflare account](https://dash.cloudflare.com/sign-up).
```

    2. Set up your R2 bucket:

>        - Navigate to R2 Object Storage in your dashboard

Reviewer: ```suggestion
       - Navigate to R2 Object Storage in your dashboard. You may need to provide your credit-card information.
```

       - Click "Create bucket"
       - Enter a unique bucket name
       - Select "Automatic" for location
       - Choose "Standard" for storage class
       - Confirm by clicking "Create bucket"
    3. Configure public access:

>        - Open your bucket's Settings

Reviewer: ```suggestion
       - On your bucket's dashboard, click on the Settings tab.
```

       - Locate the Public Access section

>        - Enable R2.dev subdomain access

Reviewer: ```suggestion
       - Scroll down to the Public Access section, and click on "Allow Access" in the "R2.dev subdomain" card.
```

       - Type 'allow' to confirm
       - Copy the Public R2.dev Bucket URL for your `S3_FILE_URL`

>     4. Set up your credentials:

Reviewer: ```suggestion
    4. Retrieve credentials:
```

>        - Find your Account ID in the R2 dashboard's top right corner

Reviewer: Would remove this point and instead add it to the [options table](https://docs.medusajs.com/resources/architectural-modules/file/s3#s3-file-module-options) similar to other platforms. Add for `endpoint` the explanation on how to set it for Cloudflare

Reviewer: I just realized you added it to the table, so just include the instructions of where you can find the account ID in there.

>        - [Generate API tokens](https://dash.cloudflare.com/?to=/:account/r2/api-tokens):

Reviewer: ```suggestion
       - [Go to API tokens page](https://dash.cloudflare.com/?to=/:account/r2/api-tokens):
```

>          - Select **Create API token**

Reviewer: ```suggestion
         - Select "Create API token"
```

>          - Edit the **R2 Token** name

Reviewer: ```suggestion
         - Edit the "R2 Token" name
```

>          - Under **Permissions**, select appropriate permission types

Reviewer: Can you specify here the appropriate permission type? I think it should be "Object Read & Write"

Also, remove the bold on "Permissions"

>          - Optionally scope token to specific buckets for Object Read/Write permissions

Reviewer: ```suggestion
         - You can optionally specify the buckets that this API token has access to under the "Specify bucket(s)" section.
```

>          - Click **Create API Token**

Reviewer: ```suggestion
         - Once done, click the "Create API Token" button.
```

>          - Save the generated token as your **`S3_ACCESS_KEY_ID`, `S3_SECRET_ACCESS_KEY`**

Reviewer: Actually you get an access key ID and secret access key after creating the token. Let's instruct the user to use that instead.

```suggestion
         - You'll receive an access key ID and a secret access key. Save them to use them later for the `S3_ACCESS_KEY_ID` and `S3_SECRET_ACCESS_KEY` environment variables.
```

       - Record your bucket name for configuration
       - Save your public bucket URL for reference
    
       ```scheme
       S3_FILE_URL="https://<Public-R2.dev-Bucket-URL-Generated-on-step-3"
       S3_ACCESS_KEY_ID="generated-on-step-4"
       S3_SECRET_ACCESS_KEY="generated-on-step-4"
       S3_REGION="auto"
       S3_BUCKET="your-bucket-name"
       S3_ENDPOINT="https://<your-account-id>.r2.cloudflarestorage.com"

>        ```

Reviewer: Remove
